File: src/net/liplum/registries/ContentRegistry.java
package net.liplum.registries;

import mindustry.ctype.ContentList;

import java.util.LinkedList;
import java.util.List;

public class ContentRegistry {
    public static List<ContentList> ContentLists = new LinkedList<>();
    public static ContentList ItemList = with(new CioItems());
    public static ContentList BlockList = with(new CioBlocks());

    public static ContentList with(ContentList cl) {
        ContentLists.add(cl);
        return cl;
    }

    public static void loadContent() {
        for (ContentList cl : ContentLists) {
            cl.load();
        }
    }
}

File: main/src/net/liplum/registries/ContentRegistry.kt
package net.liplum.registries

import java.util.*

object ContentRegistry {
    var ContentTables: MutableList<ContentTable> = LinkedList()
    var ItemList = CioItems.add()
    var LiquidList = CioLiquids.add()
    var StatusEffectList = CioSEffects.add()
    var BulletsList = CioBulletTypes.add()
    var BlockList = CioBlocks.add()
    var UnitTypeList = CioUnitTypes.add()
    @JvmStatic
    fun loadContent() {
        ContentTables.forEach {
            it.firstLoad()
        }
        ContentTables.forEach {
            it.load()
        }
        ContentTables.forEach {
            it.lastLoad()
        }
        CioTechTree.loadSerpulo()
    }
    @JvmStatic
    fun ContentTable.add(): ContentTable {
        ContentTables.add(this)
        return this
    }
}